pkg/trace/obfuscate/sql_tokenizer.go

```diff
@@ -278,12 +286,23 @@ func (tkn *SQLTokenizer) Scan() (TokenKind, []byte) {
 			}
 			return TokenKind(ch), tkn.bytes()
 		case '!':
-			if tkn.lastChar == '=' {
+			switch tkn.lastChar {
+			case '=':
 				tkn.advance()
 				return NE, []byte("!=")
+			case '~':
+				tkn.advance()
+				switch tkn.lastChar {
```

Comment:
wait, so this is a switch on `tkn.lastChar` inside a switch on `tkn.lastChar`?

Reply:
oh nevermind I see this now https://github.com/DataDog/datadog-agent/blob/22af68f8127615594ed6e781fa93a472b38cef83/pkg/trace/obfuscate/sql_tokenizer.go#L684

Reply:
might be worth a comment here?

Reply:
I could add a comment but we actually do it [fairly](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/obfuscate/sql_tokenizer.go#L237-L247) [often](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/obfuscate/sql_tokenizer.go#L262-L270) in the tokenizer so this might be overkill to add a comment on those usages.

Reply:
oh I gotcha 👍 

just looks weird to me